exonum-java-binding-common/src/main/java/com/exonum/binding/common/serialization/ZonedDateTimeSerializer.java

```diff
@@ -17,22 +17,43 @@
 
 package com.exonum.binding.common.serialization;
 
-import sun.reflect.generics.reflectiveObjects.NotImplementedException;
+import static com.exonum.binding.common.serialization.SerializationUtils.checkLength;
+
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.time.Instant;
+import java.time.ZoneOffset;
 import java.time.ZonedDateTime;
 
 enum ZonedDateTimeSerializer implements Serializer<ZonedDateTime> {
   INSTANCE;
 
+  private static final int BYTE_BUFFER_CAPACITY = Long.BYTES + Integer.BYTES;
+
   @Override
   public byte[] toBytes(ZonedDateTime value) {
-    // TODO: implement
-    throw new NotImplementedException();
+    long seconds = value.toEpochSecond();
+    int nanos = value.getNano();
+    ByteBuffer buffer = ByteBuffer.allocate(BYTE_BUFFER_CAPACITY)
+        .order(ByteOrder.LITTLE_ENDIAN);
+    buffer.putLong(seconds);
+    buffer.putInt(nanos);
+    return buffer.array();
   }
 
   @Override
   public ZonedDateTime fromBytes(byte[] serializedValue) {
-    // TODO: implement
-    throw new NotImplementedException();
+    checkLength(serializedValue, BYTE_BUFFER_CAPACITY);
+
+    ByteBuffer buffer = ByteBuffer.wrap(serializedValue)
+        .order(ByteOrder.LITTLE_ENDIAN);
+    return retrieveZdtFromBuffer(buffer);
   }
 
+  private ZonedDateTime retrieveZdtFromBuffer(ByteBuffer buffer) {
+    long seconds = buffer.getLong();
+    int nanos = buffer.getInt();
+    Instant instant = Instant.ofEpochSecond(seconds, nanos);
+    return ZonedDateTime.ofInstant(instant, ZoneOffset.UTC);
```

Comment:
Is this serializer publicly available (through a method, as it is package-private)? If so, the fact that it _ignores_ time zone during serialization does not seem right _in a general-purpose_ date time serializer. 

I think we shall either
1. Change it to the serializer of Instant (and convert instant from core to utc somewhere else).
2. Make it internal, change the name to UtcZonedDTS, add a check that passed DTS is indeed in UTC.

The first might still require a serializer, because we provide direct access to indices (cannot convert anywhere but in serializer), hence the latter seems to be the path to take.